src/webui/webapplication.cpp

```diff
@@ -774,6 +777,33 @@ void WebApplication::action_command_bottomPrio()
     BitTorrent::Session::instance()->bottomTorrentsPriority(hashes);
 }
 
+void WebApplication::action_command_setLocation()
+{
+    CHECK_URI(0);
+    CHECK_PARAMETERS("hashes" << "location");
+
+    QStringList hashes = request().posts["hashes"].split("|");
+    QString newLocationPath = request().posts["location"].trimmed();
+
+    // check location exists
+    QFileInfo newLocation(newLocationPath);
+    if (!newLocation.exists()) {
+        Logger::instance()->addMessage(tr("Invalid path '%1'").arg(newLocationPath));
+        return;
+    }
+
+    foreach (const QString &hash, hashes) {
+        BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
+        if (torrent) {
+            // get old location
+            const QString oldLocation = torrent->savePath();
+            Logger::instance()->addMessage(tr("Moving torrent '%1' to '%2'").arg(torrent->hash()).arg(newLocationPath));
```

Comment:
IMO, the msg in `TransferListWidget::setSelectedTorrentsLocation();` is pretty clear, you should use it.
Also, you should add "WebUI" in the msg, so users can know where it comes from, ex:
`WebUI Set location: moving \"%1\", from \"%2\" to \"%3\"`